crates/router/tests/connectors/rapyd.rs

```diff
@@ -0,0 +1,233 @@
+            off_session: None,
+            setup_mandate_details: None,
+            capture_method: None,
+            browser_info: None,
+            order_details: None,
+            email: None,
+        },
+        response: Err(types::ErrorResponse::default()),
+        payment_method_id: None,
+        address: PaymentAddress::default(),
+        connector_meta_data: None,
+        amount_captured: None,
+    }
+}
+
+fn construct_refund_router_data<F>() -> types::RefundsRouterData<F> {
+    let auth = ConnectorAuthentication::new()
+        .rapyd
+        .expect("Missing Rapyd connector authentication configuration");
+
+    types::RouterData {
+        flow: PhantomData,
+        connector_meta_data: None,
+        merchant_id: "rapyd".to_string(),
+        connector: "rapyd".to_string(),
+        payment_id: uuid::Uuid::new_v4().to_string(),
+        attempt_id: None,
+        status: enums::AttemptStatus::default(),
+        router_return_url: None,
+        payment_method: enums::PaymentMethodType::Card,
+        auth_type: enums::AuthenticationType::NoThreeDs,
+        connector_auth_type: auth.into(),
+        description: Some("This is a test".to_string()),
+        return_url: None,
+        request: types::RefundsData {
+            amount: 100,
+            currency: enums::Currency::USD,
+            refund_id: uuid::Uuid::new_v4().to_string(),
+            connector_transaction_id: String::new(),
+            refund_amount: 10,
+            connector_metadata: None,
+            reason: None,
+        },
+        response: Err(types::ErrorResponse::default()),
+        payment_method_id: None,
+        address: PaymentAddress::default(),
+        amount_captured: None,
+    }
+}
+
+#[actix_web::test]
+#[ignore]
+async fn test_rapyd_payment_success() {
+    use router::{configs::settings::Settings, connector::Rapyd, services};
+
+    let conf = Settings::new().unwrap();
+    static CV: Rapyd = Rapyd;
+    let connector = types::api::ConnectorData {
+        connector: Box::new(&CV),
+        connector_name: types::Connector::Checkout,
+        get_token: types::api::GetToken::Connector,
+    };
+    let state = routes::AppState::with_storage(conf, StorageImpl::PostgresqlTest).await;
+    let connector_integration: services::BoxedConnectorIntegration<
+        '_,
+        types::api::Authorize,
+        types::PaymentsAuthorizeData,
+        types::PaymentsResponseData,
+    > = connector.connector.get_connector_integration();
+    let request = construct_payment_router_data();
+
+    let response = services::api::execute_connector_processing_step(
+        &state,
+        connector_integration,
+        &request,
+        payments::CallConnectorAction::Trigger,
+    )
+    .await
+    .unwrap();
+
+    println!("{response:?}");
+
+    assert!(
+        response.status == enums::AttemptStatus::Charged,
+        "The payment failed"
+    );
+}
+
+#[actix_web::test]
+#[ignore]
+async fn test_rapyd_refund_success() {
```

Comment:
Why has this test been ignored?

Reply:
corrected!

Reply:
Could you also add a screenshot of output for these tests?

`cargo test --package router --test connectors -- rapyd`

Reply:
Done